src/backend/distributed/transaction/distributed_deadlock_detection.c

```diff
@@ -25,11 +25,247 @@
 #include "utils/timestamp.h"
 
 
+static bool CheckDeadlockForDistributedTransaction(TransactionNode *transactionNode);
+static void ResetVisitedFields(HTAB *adjacencyList);
+static List * GetSortedDistributedTransactionIdList(HTAB *adjacencyList);
+static void AssicateDistributedTransactionWithBackendProc(TransactionNode *
+														  transactionNode);
 static TransactionNode * GetOrCreateTransactionNode(HTAB *adjacencyList,
 													DistributedTransactionId *
 													transactionId);
 static uint32 DistributedTransactionIdHash(const void *key, Size keysize);
-static int DistributedTransactionIdCompare(const void *a, const void *b, Size keysize);
+static int DistributedTransactionIdCompareHash(const void *a, const void *b, Size
+											   keysize);
+static int DistributedTransactionIdCompare(const void *a, const void *b);
+
+
+PG_FUNCTION_INFO_V1(check_distributed_deadlocks);
+
+
+/*
+ * check_distributed_deadlocks is the external API for manually
+ * checking for distributed deadlocks. For the details, see
+ * CheckForDistributedDeadlocks().
+ */
+Datum
+check_distributed_deadlocks(PG_FUNCTION_ARGS)
```

Comment:
Nah, I was joking. I like the name (FWIW, it's a jokey reference to both Woody Guthrie's Guitar sticker and the Laundry files book series), but it's probably not quite appropriate...